components/tidb_query/src/rpn_expr/impl_cast.rs

```diff
@@ -2137,4 +2313,1249 @@ mod tests {
             check_warning(&ctx, err_code, log.as_str());
         }
     }
+
+    /// base_cs
+    ///   - (cast_func_input, in_union, is_res_unsigned, base_result)
+    ///   - the base_result is the result **should** produce by
+    /// the logic of cast func above `produce_dec_with_specified_tp`
+    fn test_as_decimal_helper<T: Clone, FnCast, FnToStr>(
+        base_cs: Vec<(T, bool, bool, Decimal)>,
+        cast_func: FnCast,
+        input_as_debug_str_func: FnToStr,
+        func_name: &str,
+    ) where
+        FnCast: Fn(&mut EvalContext, &RpnFnCallExtra, &Option<T>) -> Result<Option<Decimal>>,
+        FnToStr: Fn(&T) -> String,
+    {
+        #[derive(Clone, Copy, Debug)]
+        #[allow(clippy::enum_variant_names)]
+        enum Cond {
+            TargetIntPartLenLessThanOriginIntPartLen,
```

Comment:
Could we shorten these names?

Reply:
If shorten it, then I need to add doc for it. I think add doc is not a good idea.
There is may long function name in TiKV.